plotille/_figure.py

```diff
@@ -324,24 +363,39 @@ def height_vals(self):
         return self.Y
 
     def write(self, canvas, with_colors, in_fmt):
+
+        # plot all points with optional text first, then lines
+        # separating out points from lines allows single-point scatters to successfuly plot
+
+        # make point iterators
+        color = self.lc if with_colors else None
+
+        points = zip(map(in_fmt.convert, self.X), map(in_fmt.convert, self.Y))
+
+        for index, (x, y) in enumerate(points):
+
+            if(isinstance(self.text, list) and index <= len(self.text)):
+                text_for_point = self.text[index]
+            else:
+                text_for_point = self.text
+
+            canvas.point(x, y, color=color, overlay=self.overlay, marker=self.marker, text=text_for_point)
+
         # make point iterators
         from_points = zip(map(in_fmt.convert, self.X), map(in_fmt.convert, self.Y))
         to_points = zip(map(in_fmt.convert, self.X), map(in_fmt.convert, self.Y))
 
         # remove first point of to_points
         next(to_points)
 
-        color = self.lc if with_colors else None
         # plot points
         for (x0, y0), (x, y) in zip(from_points, to_points):
-            canvas.point(x0, y0, color=color)
-
-            canvas.point(x, y, color=color)
             if self.interp == 'linear':
                 canvas.line(x0, y0, x, y, color=color)
```

Comment:
Can it be, that the line interpolations overwrite the text from above?

Also, if the plotting of points is moved to the code above, I would put all the code from 384ff into the `if self.interp == 'linear':` block.